### PR TITLE
feat(reactions): added new sructures, managers and events

### PR DESCRIPTION
**Pull request by DjDeveloperr**
This PR introduces
- Structures: `MessageReaction`, changed `Message`, `CommandCategory` and `CategoriesManager`.
- Managers: `MessageReactionsManager`, `ReactionUsersManager`
- Events
  - [x] `messageReactionAdd`
  - [x] `messageReactionRemove`
  - [x] `messageReactionRemoveEmoji`
  - [x] `messageReactionRemoveAll`
  - ...and types for these events for intellisense left, will add commits to this PR and edit this
- Introducing Command Categories - these allow to have grouped settings for commands, adding up a `CommandClient#categories` and `Command#category`. Plus a `CommandBuilder` for those who want to build commands without classes.

### CODE REVIEW

**Review by Helloyunho — APPROVED** [review]
Great! I'll merge this now.

File: src/gateway/handlers/presenceUpdate.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { Gateway, GatewayEventHandler } from '../index.ts'
+
+export const presenceUpdate: GatewayEventHandler = async (
+  gateway: Gateway,
+  d: any
+) => {}
```
Comment (AkiaCode) [inline src/gateway/handlers/presenceUpdate.ts:6]: what is it?

Comment (DjDeveloperr) [reply src/gateway/handlers/presenceUpdate.ts:6]: Empty event for now, was going to work on it later.

Comment (AkiaCode) [reply src/gateway/handlers/presenceUpdate.ts:6]: Ok

**Pull request merged**